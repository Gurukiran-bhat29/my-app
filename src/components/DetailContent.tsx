import { useState, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import { Article } from '../utils/article';
import { IndividualUrl } from '../utils/constants';
import { useSearchParams } from "react-router-dom";
import ErrorBoundary from './ErrorBoundary';

const initialVal = {
  id: '',
  title: '',
  summary: '',
  fullText: ''
}

function DetailContent() {
  const [selectedArticle, setSelectedArticles] = useState<Article>(initialVal);
  const [error, setError] = useState(false);

  const [searchParam] = useSearchParams();
  const navigate = useNavigate();

  const showContent = async (id: any) => {
    try {
      const res = await fetch(IndividualUrl + id);

      if (!res.ok) {
        setError(true);
      }

      const jsonRes = await res.json();
      setSelectedArticles(jsonRes)
    } catch (error) {
      console.log('Error: ', error);
    }
  }

  useEffect(() => {
    showContent(searchParam.get('v'));
  }, [searchParam])

  if (error) {
    return <ErrorBoundary />
  }

  const { title, summary, fullText } = selectedArticle;


  return (
    <div className='content-container'>
      {title && (
        <>
          <div className='detail-content'>
            <h1>{title}</h1>
            <h3>{summary}</h3>
            <p className='full-text'>{fullText}</p>
          </div>
          <button className='back-button' onClick={() => navigate('/')}>Back</button>
        </>
      )}
    </div>
  );
}

export default DetailContent;
