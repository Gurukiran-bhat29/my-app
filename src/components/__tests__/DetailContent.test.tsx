import { render, screen, waitFor } from "@testing-library/react"
import DetailContent from "../DetailContent"
import { BrowserRouter } from "react-router-dom"
import MOCK_DATA from '../../MockData/article.json';

beforeEach(() => {
  jest.clearAllMocks();
  global.fetch = jest.fn(() =>
    Promise.resolve({
      ok: true,
      json: () =>
        Promise.resolve(MOCK_DATA),
    })
  ) as jest.Mock;
});

it('should render article details with back button', async () => {
  render(
    <BrowserRouter>
      <DetailContent />
    </BrowserRouter>
  )

  expect(screen.getByTestId('article-contents'));

  const button = await waitFor(() => screen.getByRole("button"));

  expect(button).toBeInTheDocument();
})