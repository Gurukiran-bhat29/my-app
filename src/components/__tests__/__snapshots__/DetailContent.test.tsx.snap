// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render article details with back button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="content-container"
        data-testid="article-contents"
      >
        <div
          class="detail-content"
        >
          <h1>
            Exploring the Wonders of the Great Barrier Reef
          </h1>
          <h3>
            Discover the incredible biodiversity and natural beauty of the Great Barrier Reef.
          </h3>
          <p
            class="full-text"
          >
            The Great Barrier Reef, stretching over 2,300 kilometers along the coast of Queensland, Australia, is a marine wonderland. Home to thousands of species of fish, coral, and marine life, it is the largest coral reef system in the world. Snorkelers and divers flock here to witness its vibrant colors and diverse ecosystems. However, it faces threats from climate change and pollution, making conservation efforts crucial to its preservation.
          </p>
        </div>
        <button
          class="back-button"
        >
          Back
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="content-container"
      data-testid="article-contents"
    >
      <div
        class="detail-content"
      >
        <h1>
          Exploring the Wonders of the Great Barrier Reef
        </h1>
        <h3>
          Discover the incredible biodiversity and natural beauty of the Great Barrier Reef.
        </h3>
        <p
          class="full-text"
        >
          The Great Barrier Reef, stretching over 2,300 kilometers along the coast of Queensland, Australia, is a marine wonderland. Home to thousands of species of fish, coral, and marine life, it is the largest coral reef system in the world. Snorkelers and divers flock here to witness its vibrant colors and diverse ecosystems. However, it faces threats from climate change and pollution, making conservation efforts crucial to its preservation.
        </p>
      </div>
      <button
        class="back-button"
      >
        Back
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
